### PR TITLE
17 aug next item

### PR DESCRIPTION
**Pull request by yakovhaiilo**

### CODE REVIEW

**Review by nirnaor — CHANGES_REQUESTED** [review]
Some of the changes requested in the [original pull request](https://github.com/wix/pro-gallery/pull/534#discussion_r691869046) were  not fixed :( 

1. 😞  scrollToGroup [call](https://github.com/wix/pro-gallery/blob/75e26d80b8f5d9a7ebacf8d63d2f8d30fd64d1f0/packages/gallery/src/components/gallery/proGallery/slideshowView.js#L334-L342) is duplicated with scrollToItem [call](https://github.com/wix/pro-gallery/blob/75e26d80b8f5d9a7ebacf8d63d2f8d30fd64d1f0/packages/gallery/src/components/gallery/proGallery/slideshowView.js#L283-L291)
2. 😄  
3. 😄  using `onScroll`
4. 😄 
5. 😞  [This](https://github.com/wix/pro-gallery/blob/75e26d80b8f5d9a7ebacf8d63d2f8d30fd64d1f0/packages/gallery/src/components/gallery/proGallery/slideshowView.js#L311-L315) is a duplication of [this](https://github.com/wix/pro-gallery/blob/75e26d80b8f5d9a7ebacf8d63d2f8d30fd64d1f0/packages/gallery/src/components/gallery/proGallery/slideshowView.js#L345-L349)

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 1]
```diff
@@ -208,193 +226,187 @@ class SlideshowView extends GalleryComponent {
...
+    let nextIndex;
+    if (initiator === 'nextItem') {
+      if (ignoreScrollPosition) {
+        nextIndex = this.state.activeIndex;
+      } else {
+        if (
+          avoidIndividualNavigation &&
+          !(this.props.styleParams.groupSize > 1)
+        ) {
+          nextIndex = this.getCenteredItemIdxByScroll();
```
Comment (nirnaor) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: @yakovhaiilo Can you reuse this line and line 268?

Comment (nirnaor) [reply packages/gallery/src/components/gallery/proGallery/slideshowView.js]: This behaviour is duplicated: 
`nextIndex = this.getCenteredGroupIdxByScroll() + direction;`
It is used twice: 
1. `initiator === 'nextGroup`
2. `initiator === 'nextItem && !ignoreScrollPosition && (avoidIndividualNavigation &&
          !(this.props.styleParams.groupSize > 1))`

So you can make one if at the top level of your function that handles that (and `returns`)
Afterwards you deal with the complicated `nextItem` behaviour

**Review by nirnaor — CHANGES_REQUESTED** [review]

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 2]
```diff
@@ -208,193 +226,187 @@ class SlideshowView extends GalleryComponent {
...
+
+      if (!this.props.styleParams.slideshowLoop) {
+        nextIndex = Math.min(
+          this.props.galleryStructure.items.length - 1,
+          nextIndex
+        );
+        nextIndex = Math.max(0, nextIndex);
+      }
+    } else if (initiator === 'nextGroup') {
+      nextIndex = this.getCenteredGroupIdxByScroll() + direction;
```
Comment (nirnaor) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: These functions are duplicated 
`getCenteredItemIdxByScroll`
`getCenteredGroupIdxByScroll`

They are doing the same thing with different variable names. Please use one of them and delete the other function

**Review by nirnaor — CHANGES_REQUESTED** [review]

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js (hunk 2 above)
Comment (yakovhaiilo) [reply packages/gallery/src/components/gallery/proGallery/slideshowView.js]: @nirnaor It's not exactly the same thing, but I deleted one of them and combined the different logics

Comment (nirnaor) [reply packages/gallery/src/components/gallery/proGallery/slideshowView.js]: ok @yakovhaiilo let me know when I can review again

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 3]
```diff
@@ -208,193 +226,188 @@ class SlideshowView extends GalleryComponent {
...
     avoidIndividualNavigation,
     ignoreScrollPosition,
-    isContinuousScrolling
+    isContinuousScrolling,
+    scrollingUpTheGallery,
   }) {
     if (this.isSliding) {
       return;
     }
     this.isSliding = true;
```
Comment (nirnaor) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: The lines: `    this.isAutoScrolling = true` and `    this.isSliding = true;` are duplicated in `nextGroup` and `nextItem`. 
You can remove the duplicated code by doing the following:

1. Move `this.isSliding = true` to the beginning of `getNextItemOrGroupToScrollTo` 
2. Move `this.isAutoScrolling = true` to the end of `getNextItemOrGroupToScrollTo`

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 4]
```diff
@@ -208,193 +226,188 @@ class SlideshowView extends GalleryComponent {
...
-    }
+    let nextItem = this.getNextItemOrGroupToScrollTo(
+      'nextItem',
+      direction,
+      ignoreScrollPosition,
+      avoidIndividualNavigation,
+      isAutoTrigger
+    );
 
     const { scrollToItem } = this.props.actions;
```
Comment (nirnaor) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: `scrollToItem` has no use other than the call to `this.scrollTo`
You can remove it and pass the `this.props.actions.scrollToItem` directly to `this.scrollTo`
Do the same in `nextGroup` as well. There's no need to define an extra `const` if it is used only once.
It makes it harder to see the differences between the two logics.

**Review by nirnaor — CHANGES_REQUESTED** [review]

**Review by nirnaor — APPROVED** [review]

**Comment by nirnaor** [comment]
@yakovhaiilo any reason that we wait with this merge?

**Pull request merged**